src/main/java/network/brightspots/rcv/DominionCvrReader.java

```diff
@@ -123,10 +123,14 @@
     return contests;
   }
 
-  // parse Cvr json into CastVoteRecord objects and add them to the input list
-  // (If contestId is specified, we'll only load CVRs for that contest.)
+  // parse CVR JSON for records matching the specified contestId into CastVoteRecord objects and add
+  // them to the input list
   void readCastVoteRecords(List<CastVoteRecord> castVoteRecords, String contestId)
       throws CvrParseException, UnrecognizedCandidatesException {
+    if (contestId == null) {
```

Comment:
This already covered in `passesBasicCvrSourceValidation()` so I don't think there should be any need to do it here (which I suppose means you'd revert basically everything in this file).

Reply:
I meant this more like an assertion (guarding against a coding error, not a user error), but yeah, I guess we don't really do that elsewhere, so I'll remove it. My other changes in this file still make sense, though.